File: Python/problem1119.py
class Solution:
    def removeVowels(self, S: str) -> str:
        s = set(['a','e','i','o','u'])
        ans = []
        for s0 in S:
            if s0 not in s: #在集合中使用in是O(1)的时间复杂度
                ans.append(s0)
        # S = ans
        return ''.join(ans)

File: C++/problem1119.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <map>
#include <set>
#include <algorithm>
#include <cmath>
#include <ctime>
#include <cstdlib>
#include <windows.h>

using namespace std;

class Solution {
public:
    string removeVowels(string S) {
        set<char> s {'a', 'e', 'i', 'o', 'u'};
        string ans;
        for (int k = 0; k < S.size(); ++k)
        {
            if (!s.count(S[k]))
            {
                ans.push_back(S[k]);
            }
        }
        return ans;
    }
};

int main()
{
    string S = "leetcodeisacommunityforcoders";
    Solution solu;
    string ans = solu.removeVowels(S);
    cout << ans << endl;
    return 0;
}